chips_development.md

```diff
@@ -29,12 +29,34 @@ finding the entry as easy as possible.
 
 ## Testing
 
+### Unit tests
+
 Use **Mockito** for mocking classes.
 Test classes using Easymock should be converted when worked on.
 
 Use **org.hamcrest.MatcherAssert.assertThat** instead of
 AssertTrue where that makes the code clearer.
 
+When working with CHIPS Rules please use the assertions provided of appropriate.
+
+The assert methods run the rule as represented by the testRule instance variable and
+check the number of issues found.
+
+| Method                                        | Description                                                                                      |
+|-----------------------------------------------|--------------------------------------------------------------------------------------------------|
+| assertExpectedIssuesMatches(int expectedIssues) | Run testRule and assert that the number of issues generated matches the value of expectedIssues. |
+| assertZeroIssues()	                           | Run testRule and assert that the number of issues generated is 0.                                |
+| assertOneOrMoreIssues()                       | Run testRule and assert that the number of issues generated is > 0.                              |
+
+### Integration tests
+
+Integration tests for Data Access Object (DAO) classes that interact with the database
+are required.
```

Comment:
Perhaps add 'these are located in the server/test/it/src directory and are subclasses of AbstractSpringContextPersistenceTest' ?